File: app/src/main/java/tv/spacedentist/android/util/SDLogger.java
package tv.spacedentist.android.util;

import android.util.Log;

/**
 * A wrapper around {@link Log} that can be mocked and injected in tests.
 */
public interface SDLogger {
    void i(String tag, String msg);
    void d(String tag, String msg);
    void e(String tag, String msg);
    void e(String tag, String msg, Throwable tr);
}

File: app/src/main/java/tv/spacedentist/android/util/SDLogger.kt
package tv.spacedentist.android.util

import android.util.Log

/**
 * A wrapper around [Log] that can be mocked and injected in tests.
 */
interface SDLogger {
    fun i(tag: String, msg: String) {}
    fun d(tag: String, msg: String) {}
    fun e(tag: String, msg: String) {}
    fun e(tag: String, msg: String, tr: Throwable) {}
}
